adal/src/main/java/com/microsoft/aad/adal/PRNGFixes.java

```diff
@@ -249,22 +244,17 @@ protected void engineNextBytes(byte[] bytes) {
                 // Mix in the device- and invocation-specific seed.
                 engineSetSeed(generateSeed());
             }
-            DataInputStream in = null;
+
             try {
-                in = getUrandomInputStream();
+                DataInputStream in;
                 synchronized (SLOCK) {
+                    in = getUrandomInputStream();
+                }
+                synchronized (in) {
```

Comment:
just curious: why was readFully moved from being protected by SLOCK to being protected by the input stream's object?